aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -178,6 +180,28 @@ module aptos_framework::block {
         );
     }
 
+    ///  Emit a `NewBlockEvent` event. This function will be invoked by write set script directly to generate the
+    ///  new block event for WriteSetPayload.
+    public fun emit_writeset_block_event(framework_signer: &signer, fake_block_hash: address) acquires BlockResource {
```

Comment:
why use a framework_signer instead of vm_signer? I want to differentiate this from governance proposal such that this can only be used offchain, not onchain

Reply:
The issue is that you need framework signer to trigger reconfigure and other functionalities of the writeset script. So a vm_signer can only give you very limited capability.

Reply:
e.g: publish new aptos framework/removing validator sets.

Reply:
let's at least have vm_signer here as well to prevent onchain access

Reply:
The problem was that the writeset script now can only have one signer. So it's either the vm signer or the framework signer unfortunately

Reply:
hmm, can we pass the vm_signer here? https://github.com/aptos-labs/aptos-core/blob/b44af464fdea8fa7ef65b171081050e2da6c3232/aptos-move/aptos-vm/src/aptos_vm.rs#L753